spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-config-sample/README.adoc

```diff
@@ -0,0 +1,78 @@
+
+== Setup
+
+1. Configure your GCP project ID and credentials by following link:../../docs/src/main/asciidoc/core.adoc#project-id[these instructions].
++
+Alternatively, if you have the https://cloud.google.com/sdk/[Google Cloud SDK] installed and initialized, and are logged in with https://developers.google.com/identity/protocols/application-default-credentials[application default credentials], Spring will auto-discover those parameters for you.
+
+2. Make sure that GCP Cloud Pub/Sub API is enabled.
+
+== Running the code sample with configuration derived from local filesystem
+
+1. In separate terminal windows, start the `spring-cloud-gcp-pubsub-config-sample-server` and `spring-cloud-gcp-pubsub-config-sample-client` applications with the `$ mvn spring-boot:run` command in the same folder as the apps' `pom.xml` files.
+
+2. Visit http://localhost:8080/message in a browser; observe the default value of `none` printed on the page.
+This happens because there is no configuration available for config server yet.
+
+3. Create a temporary directory `/tmp/config`.
+Alternatively, update `spring.cloud.config.server.native.searchLocationsconfig` property in configuration server's link:spring-cloud-gcp-pubsub-config-sample-server/src/main/resources/application.properties[application.properties] with the directory you will use, and restart the config server.
+
+4. Place a file named `application.properties` with the following contents into the directory above:
++
+----
+example.message = message from the local filesystem
+----
+
+5. Notify the configuration server that new configuration is available.
+In a deployed server, this would be done through a GitHub webhook, or similar mechanism.
++
+----
+curl -X POST http://localhost:8888/monitor -d 'path=application'
+----
+
+6. Visit http://localhost:8080/message in a browser; observe the value now matches the value updated in the file system.
+
+== Running the code sample with configuration maintained in GitHub
+
+Keeping configuration in the local file system is simple, but not particularly useful -- if the configuration were available on the same file system as the client application, it could just get it without having to maintain a configuration server.
+A more realistic scenario is keeping configuration under version control.
+
+1. In the root of a GitHub repository, create a file named `application.properties` containing the following line:
++
+----
+example.message = message from GitHub
+----
+
+2. In configuration server's link:spring-cloud-gcp-pubsub-config-sample-server/src/main/resources/application.properties[application.properties]:
+  * Comment out `spring.profiles.active` and `spring.cloud.config.server.native.searchLocations` properties.
+  * Uncomment `spring.cloud.config.server.git.uri` property; set its value to either HTTPS or SSH repository URI.
+  * If you opted for SSH, keys stored in `~/.ssh` should work automatically.
```

Comment:
What will happen for the HTTPS option? Will it only work with public github repos?

Reply:
I don't think github API is available without authentication at all.